pkg/fs/fs_test.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fs
+
+import (
+	"errors"
+	"os"
+	"testing"
+)
+
+func TestTempDir(t *testing.T) {
+	tests := []struct {
+		name   string
+		dir    string
+		prefix string
+		expect error
+	}{
+		{
+			name:   "all is right",
+			dir:    ".",
+			prefix: "tmp",
+			expect: nil,
+		},
+		{
+			name:   "dir is nil",
+			dir:    "",
+			prefix: "tmp",
+			expect: nil,
+		},
+		{
+			name:   "dir isn't exist",
+			dir:    "/abc",
+			prefix: "tmp",
+			expect: errors.New("isn't nil"),
+		},
+		{
+			name:   "prefix is nil",
+			dir:    ".",
+			prefix: "",
+			expect: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			name, err := TempDir(tt.dir, tt.prefix)
+			os.Remove(name)
+			if (nil == err) != (nil == tt.expect) {
+				t.Errorf("expect: %v, got: %v", tt.expect, err)
+			}
+		})
+	}
+}
+
+func TestIsAbs(t *testing.T) {
+	tests := []struct {
+		name     string
+		hostPath string
+		expect   bool
+	}{
+		{
+			name:     "is absolute path",
+			hostPath: "/home/test",
+			expect:   true,
+		},
+		{
+			name:     "isn't absolute path",
+			hostPath: "./test",
+			expect:   false,
+		},
+		{
+			name:     "isn't absolute path(~)",
+			hostPath: "~/test",
+			expect:   false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ret := IsAbs(tt.hostPath)
+			if ret != tt.expect {
+				t.Errorf("expect: %v, got: %v", tt.expect, ret)
+			}
+		})
+	}
+}
+
+func TestCopy(t *testing.T) {
+	defer os.Remove("./test.txt")
+	defer os.RemoveAll("./test")
```

Comment:
tests should not use relative paths, but instead write to https://golang.org/pkg/io/ioutil/#TempDir